src/adapter/src/coord/sequencer/inner.rs

```diff
@@ -2572,7 +2572,12 @@ impl Coordinator {
         let id_bundle = self
             .index_oracle(cluster_id)
             .sufficient_collections(&depends_on);
-        let timeline = self.validate_timeline_context(id_bundle.iter())?;
+        let mut timeline = self.validate_timeline_context(depends_on.clone())?;
```

Comment:
The switch from `id_bundle` to `depends_on` matches how peeks are handled. If `depends_on` has a constant view with a temporal filter, then `id_bundle` will be empty and `validate_timeline_context` will incorrectly return back `TimestampIndependent`.